core/src/apps/tezos/sign_tx.py

```diff
@@ -171,3 +204,28 @@ def _encode_zarith(w: bytearray, num):
             break
 
         write_uint8(w, 128 | byte)
+
+
+def _encode_proposal(w: bytearray, proposal):
+    proposal_tag = 5
+
+    write_uint8(w, proposal_tag)
+    write_bytes(w, proposal.source)
+    write_uint32_be(w, proposal.period)
+    write_uint32_be(w, _get_proposals_bytecount(proposal.proposals))
```

Comment:
having a function for this seems needless? just inline the `len(proposals) * PROPOSAL_LENGTH` here